packages/oid4vci-demo-frontend/src/pages/SSIInformationRequestPage/index.tsx

```diff
@@ -80,16 +98,26 @@ const SSIInformationRequestPage: React.FC = () => {
         const decodeBase64 = async (jwt: string, kid?: string): Promise<any> => {
             return JSON.parse(Buffer.from(jwt.split('.')[1], 'base64').toString())
         }
-        const handleCredentialSubject = (cs: ICredentialSubject & AdditionalClaims): Payload => {
-            if (!cs.firstName && !cs.lastName && !cs.emailAddress) {
-                return {}
-            }
+        const handleCredentialSubject = (cs: ICredentialSubject & AdditionalClaims, form?: DataFormRow[]): Payload => {
+            if (!form) {
+                if (!cs.firstName && !cs.lastName && !cs.emailAddress) {
+                    return {} as Record<string, string>;
+                }
 
-            return {
-                firstName: cs.firstName,
-                lastName: cs.lastName,
-                emailAddress: cs.emailAddress
+                return {
+                    firstName: cs.firstName,
+                    lastName: cs.lastName,
+                    emailAddress: cs.emailAddress
+                } as Record<string, string>;
+            }
+            const payload = transformFormConfigToEmptyObject(form);
+            for (const payloadKey in payload) {
+                if (payloadKey in cs) {
+                    // TODO: since this code is based on the manual flow, we have to revisit it for the wallet flow
```

Comment:
What do you mean with this token. This is the wallet flow right?

Reply:
yeah, since I wrote the code for manual flow, just changed a bit to match the wallet flow as well, but it needs revisiting when we're going to use it for wallet flow

Reply:
But only the first issuer has a manual flow, the other issuers use a wallet flow. So I guess this needs more work

Reply:
tested it for the wallet flow as well. it still needs more work regarding credential issuance. but for this ticket, it's done